test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,298 @@
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   0   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  ...
+         */
+
+        for (int baseRigths = 0; baseRigths < 7;
+             baseRigths++) {  // Test all the possible rights and the all the possible order of rights modification
+            for (int targetRigths = baseRigths + 1; targetRigths < 8; targetRigths++) {
+                rightsSetFromInt(baseRigths, rightsSet);
+
+                bool result = IoHelper::setRights(path, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set base rights */);
+                }
+
+                result = IoHelper::getRights(path, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get base rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    std::cout << "setted rigths (RWX): " << rightsSet.read << rightsSet.write << rightsSet.execute
+                              << " | readed rigths(RWX): " << isReadable << isWritable << isExecutable << std::endl;
+                    CPPUNIT_ASSERT(false /* Set base rights mismatch  with get base rights */);
+                }
+
+                rightsSetFromInt(targetRigths, rightsSet);
+
+                result = IoHelper::setRights(path, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set target rights */);
+                }
+
+                result = IoHelper::getRights(path, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get target rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(path, true, true, true, ioError);
+                    std::cout << "setted rigths (RWX): " << rightsSet.read << rightsSet.write << rightsSet.execute
```

Comment:
Isn't this extra logging making the tests output less readable?

Reply:
idem